source/Applications/Advanced/HandEyeCalibration/UtilizeEyeInHandCalibration/UtilizeEyeInHandCalibration.cpp

```diff
@@ -97,10 +127,45 @@ int main()
         std::cout << "Computing camera pose in robot base frame" << std::endl;
         const Eigen::MatrixXf transformBaseToCamera = transformBaseToEndEffector * transformEndEffectorToCamera;
 
-        std::cout << "Computing (picking) point in robot base frame" << std::endl;
-        const Eigen::Vector4f pointInBaseFrame = transformBaseToCamera * pointInCameraFrame;
-        std::cout << "Point coordinates in robot base frame: " << pointInBaseFrame.x() << " " << pointInBaseFrame.y()
-                  << " " << pointInBaseFrame.z() << std::endl;
+        switch(enterCommand())
+        {
+            case CommandType::cmdTransformSinglePoint:
+            {
+                // define (picking) point in camera frame
+                const Eigen::Vector4f pointInCameraFrame(81.2F, 18.0F, 594.6F, 1.0F);
+                std::cout << "Point coordinates in camera frame: " << pointInCameraFrame.x() << " "
+                          << pointInCameraFrame.y() << " " << pointInCameraFrame.z() << std::endl;
+
+                std::cout << "Computing (picking) point in robot base frame" << std::endl;
+                const Eigen::Vector4f pointInBaseFrame = transformBaseToCamera * pointInCameraFrame;
+                std::cout << "Point coordinates in robot base frame: " << pointInBaseFrame.x() << " "
+                          << pointInBaseFrame.y() << " " << pointInBaseFrame.z() << std::endl;
+                break;
+            }
+            case CommandType::cmdTransformPointCloud:
+            {
+                const auto dataFile = std::string(ZIVID_SAMPLE_DATA_DIR) + "/ZividGem.zdf";
+                std::cout << "Reading ZDF frame from file: " << dataFile << std::endl;
+                const Zivid::Frame frame = Zivid::Frame(dataFile);
+
+                std::cout << "Getting point cloud from frame" << std::endl;
+                auto pointCloud = frame.pointCloud();
+
+                std::cout << "Converting to Zivid matrixes to transform entire point cloud" << std::endl;
+                const auto transformationMatrix = eigenToZivid(transformBaseToCamera);
+                pointCloud.transform(transformationMatrix);
+
+                const auto *saveFile = "transformedPointCloud.zdf";
```

Comment:
Maybe we should call this ZividGemTransformed? Or maybe we should change ZividGem name instead to something else like: OriginalPointCloud? Maybe we should actually clarify that it has to do with eye in hand data, e.g. EyeInhandPointCloud and EyeInHandPointCloudTransformed? Or PointCloudEIH / PointCloudTransformedEIH or PointCloudEyeInHandCameraFrame / PointCloudEyeInHandRobotFrame?

Maybe SampleData should have a folder(s) HandEye/EyeInHand and then we can simplify the names to pointCloudCameraFrame / pointCloudRobotFrame? This is my favorite option.

The problem I have in any case is that we load from one folder and save to a different folder, but this we do with other samples as well, right? For example convert samples.

Reply:
Changed it to ZividGemTransformed. 

We save point clouds in different folders than what we load from in all of our samples. 

I think to structure it as folders, e.g. HandEye/EyeInHand and HandEye/EyeToHand in Sample data makes sense.